taf/auth_repo.py

```diff
@@ -25,36 +24,60 @@ class AuthRepoMixin(TAFRepository):
     _conf_dir = None
 
     def __init__(
-        self, conf_directory_root, out_of_band_authentication=None, hosts=None
+        self,
+        path,
+        name=None,
+        urls=None,
+        custom=None,
+        default_branch="master",
+        conf_directory_root=None,
+        out_of_band_authentication=None,
+        hosts=None,
+        *args,
+        **kwargs,
     ):
+        """
+        Args:
+          path: repository's full filesystem path or path to the root directory which contains the repository
+          name (optional): repository's name, which is appended to the root directory to form the full path.
+          Must be in the namespace/name format. If not specified, name will be determined based repository's
+          full path.
+          urls (optional): repository's urls
+          custom (optional): a dictionary containing other data
+          default_branch (optional): repository's default branch ("master" if not defined)
```

Comment:
Since we have not updated our repositories yet, I'd prefer to focus on the pull rework first and then deal with the name change. Unless you think that this is urgent, I vote for creating an issue and dealing with it later.